docs/guide/acquiring_data/helioviewer.rst

```diff
@@ -18,64 +18,104 @@ install Glymur on your system.
+the API using the same parameters one would use when making a web request. Note that
+HelioviewerClient is not meant to offer full access to the HelioViewer API and 
+as such it does not support all the inputs that the official API allows. But you can pass
+pass in extra keywords to communicate with the API. We cannot guarantee that the undocumented 
+keywords will work, nor will we provide support for those.
 
 Nearly all requests require the user to specify the data they are interested in
-and this can be done using one of two methods:
-
-1. Call "get_data_sources()" to get a list of the data that is available, and use the source id numbers referenced in the result to refer to a particular dataset, or,
-2. Specify the four components of a Helioviewer.org data source or layer: *observatory*, *instrument*, *detector* and *measurement*.
-
-Let's begin by getting a list of data sources available on the server
-using the get_datasources method::
-
-    >>> from sunpy.net.helioviewer import HelioviewerClient
-
-    >>> hv = HelioviewerClient()
-    >>> datasources = hv.get_data_sources()  # doctest: +REMOTE_DATA
-
-    >>> # print a list of datasources and their associated ids
-    >>> for observatory, instruments in datasources.items():  # doctest: +REMOTE_DATA
-    ...     for inst, detectors in instruments.items():  # doctest: +REMOTE_DATA
-    ...         for det, measurements in detectors.items():  # doctest: +REMOTE_DATA
-    ...             for meas, params in measurements.items():  # doctest: +REMOTE_DATA
-    ...                 print("%s %s: %d" % (observatory, params['nickname'], params['sourceId']))  # doctest: +REMOTE_DATA
-    SOHO EIT 171: 0
-    SOHO EIT 195: 1
-    SOHO EIT 284: 2
-    SOHO EIT 304: 3
-    SOHO LASCO C2: 4
-    SOHO LASCO C3: 5
-    SOHO MDI Mag: 6
-    SOHO MDI Int: 7
-    SDO AIA 94: 8
-    SDO AIA 131: 9
-    SDO AIA 171: 10
-    SDO AIA 193: 11
-    SDO AIA 211: 12
-    SDO AIA 304: 13
-    SDO AIA 335: 14
-    SDO AIA 1600: 15
-    SDO AIA 1700: 16
-    SDO AIA 4500: 17
-    SDO HMI Int: 18
-    SDO HMI Mag: 19
-    STEREO_A EUVI-A 171: 20
-    STEREO_A EUVI-A 195: 21
-    STEREO_A EUVI-A 284: 22
-    STEREO_A EUVI-A 304: 23
-    STEREO_A COR1-A: 28
-    STEREO_A COR2-A: 29
-    STEREO_B EUVI-B 171: 24
-    STEREO_B EUVI-B 195: 25
-    STEREO_B EUVI-B 284: 26
-    STEREO_B EUVI-B 304: 27
-    STEREO_B COR1-B: 30
-    STEREO_B COR2-B: 31
-    PROBA2 SWAP 174: 32
-    Yohkoh SXT AlMgMn: 33
-    Yohkoh SXT thin-Al: 34
-    Yohkoh SXT white-light: 35
+and this can be done using the following methods:
+
+1. Use the datasource_info dictionary to get the list of data source and their corresponding source id
+numbers which can be used to refer to a particular dataset or,
+2. Specify the four components of a Helioviewer.org data source or layer: *observatory*, *instrument*, 
+*detector* and *measurement*. If the *detector* is left blank then it is replaced by *None* keyword.
```

Comment:
This should be updated that any blank keywords default to None and should be changed to the correct value for the source you are interested in.